fastcrypto/src/groups/bls12381.rs

```diff
@@ -554,9 +560,14 @@ impl ToFromByteArray<GT_ELEMENT_BYTE_LENGTH> for GTElement {
                 }
             }
         }
-        match gt.in_group() {
-            true => Ok(Self::from(gt)),
-            false => Err(FastCryptoError::InvalidInput),
+
+        // We compare with gt.to_bendian() to ensure that we process a canonical representation
+        // which is uses mod p elements.
+        // TODO: Is there a more efficient way?
+        if gt.in_group() && gt.to_bendian() == *bytes {
```

Comment:
Checking that the bytes is a canonical representation of a field element may be done by comparing the bytes to a big-endian representation of the field size (1a0111ea397fe69a4b1ba7b6434bacd764774b84f38512bf6730d2a0f6b0f6241eabfffeb153ffffb9feffffffffaaab) in the loop, and if one of the coefficients is not canonical the function can fail fast. This works because the ordering on byte arrays is the lexicographical ordering which is equivalent with the natural ordering on natural when using big-endian representation.